Fix persist config values for custom components

StephanSchrader opened the pull request:
Closes #31858

Commit:
Fix persist config values for custom components

Closes #31858

Signed-off-by: Stephan Schrader <stephan.schrader@wallis.de>

ahus1 commented:
@StephanSchrader - thank you for this PR. Once the bug has been triaged, someone from the Keycloak team will look at it. Until then, there is no need to rebase the PR unless a merge conflict appears.

mposolda commented:
@StephanSchrader Thanks for the report and PR! Do you please have a chance to write also the automated test for this issue? Related tests are in `org.keycloak.testsuite.admin.ComponentsTest`, so maybe new test method can be added here? Thanks

StephanSchrader committed:
Fix persist config values for custom components

- Test case updated to cover the sending of unknown configuration keys by the client

Closes #31858

Signed-off-by: Stephan Schrader <zstephanz@gmail.com>

mposolda reviewed (approved):
@StephanSchrader Thanks
